primitives/src/channel.rs

```diff
@@ -79,11 +84,7 @@ impl FromStr for ChannelId {
     type Err = FromHexError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        if s.len() != 66 || &s[0..2] != "0x" {
-            Err(FromHexError::InvalidStringLength)
-        } else {
-            Self::from_hex(&s[2..])
-        }
+        ChannelId::from_hex(validate_channel_id(s.into())?)
```

Comment:
`s.into()` will clone the `&str` into a `String`, use the `&str` instead to avoid unnecessary cloning.